engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/analyse/FramePointerAnalysis.scala

```diff
@@ -286,9 +349,33 @@ case object FramePointerAnalysis extends IRPass {
 
   private def updateMeta(
     ir: IR,
-    framePointer: FramePointer
+    framePointer: FramePointer,
+    variableNames: Option[List[String]]
   ): Unit = {
-    ir.passData().update(this, new FramePointerMeta(framePointer))
+    ir.passData().get(this) match {
+      case None =>
+        ir.passData()
+          .update(this, new FramePointerMeta(framePointer, variableNames))
+      case Some(meta: FramePointerMeta) =>
+        val fp = if (framePointer != null) {
```

Comment:
Probably updating an `IR` with already existing data should be an (inefficiency signaling) error.

Reply:
Done in 4bcbda8 - found two IR nodes: `Case.Expr` and a `Function.Lambda` produced from it sharing the same `Metadata` - fixed too.